src/core/num_co.cc

```diff
@@ -113,10 +113,10 @@ CL_DEFUN Float_sp cl__float(Real_sp x, T_sp y) {
   case number_Ratio:
     switch (ty) {
     case number_SingleFloat:
-      x = gc::As<Real_sp>(clasp_make_single_float(clasp_to_double(x)));
+        x = gc::As<Real_sp>(clasp_make_single_float(clasp_to_double(gc::As<Number_sp>(x))));
       break;
     case number_DoubleFloat:
-      x = gc::As<Real_sp>(clasp_make_double_float(clasp_to_double(x)));
+        x = gc::As<Real_sp>(clasp_make_double_float(clasp_to_double(gc::As<Number_sp>(x))));
```

Comment:
We know x is already a Number_sp because it's passed as a Real_sp
So use gc::As_unsafe<Number_sp>(x)